smart-connector-rest-server/src/main/java/eu/knowledge/engine/rest/api/impl/ProactiveApiServiceImpl.java

```diff
@@ -46,6 +46,126 @@ public class ProactiveApiServiceImpl {
+			if (this.manager.hasSuspendedKB(knowledgeBaseId)) {
+				this.manager.removeSuspendedKB(knowledgeBaseId);
+				var response = new ResponseMessage();
+				response.setMessageType("error");
+				response.setMessage(
+						"This knowledge base has been suspended due to inactivity. Please reregister the knowledge base and its knowledge interactions.");
+				asyncResponse.resume(Response.status(Status.NOT_FOUND).entity(response).build());
+				return;
+			} else {
+				var response = new ResponseMessage();
+				response.setMessageType("error");
+				response.setMessage("Smart connector not found, because its ID is unknown.");
+				asyncResponse.resume(Response.status(Status.NOT_FOUND).entity(response).build());
+				return;
+			}
+		}
+
+		try {
+			new URI(knowledgeInteractionId);
+		} catch (URISyntaxException e) {
+			var response = new ResponseMessage();
+			response.setMessageType("error");
+			response.setMessage("Knowledge interaction not found, because its ID must be a valid URI.");
+			asyncResponse.resume(Response.status(Status.BAD_REQUEST).entity(response).build());
+			return;
+		}
+
+		if (!kb.hasKnowledgeInteraction(knowledgeInteractionId)) {
+			var response = new ResponseMessage();
+			response.setMessageType("error");
+			response.setMessage("Knowledge Interaction not found, because its ID is unknown.");
+			asyncResponse.resume(Response.status(Status.NOT_FOUND).entity(response).build());
+			return;
+		}
+
+		KnowledgeInteractionWithId ki = kb.getKnowledgeInteraction(knowledgeInteractionId);
+		if (!ki.getKnowledgeInteractionType().equals("AskKnowledgeInteraction")) {
+			var response = new ResponseMessage();
+			response.setMessageType("error");
+			response.setMessage("Given Knowledge Interaction ID should have type AskKnowledgeInteraction and not "
+					+ ki.getKnowledgeInteractionType() + ".");
+			asyncResponse.resume(Response.status(Status.BAD_REQUEST).entity(response).build());
+			return;
+		}
+
+		try {
+			var askFuture = kb.ask(knowledgeInteractionId, recipientAndBindingSetObject.recipient,
+					recipientAndBindingSetObject.bindingSet);
+
+			askFuture.thenAccept(askResult -> {
+
+				LOG.info("AskResult received, resuming async response: {}", askResult);
+				List<AskExchangeInfo> infos = askResult.getExchangeInfoPerKnowledgeBase().stream()
+						.map(aei -> new AskExchangeInfo().bindingSet(this.bindingSetToList(aei.getBindings()))
+								.knowledgeBaseId(aei.getKnowledgeBaseId().toString())
+								.knowledgeInteractionId(aei.getKnowledgeInteractionId().toString())
+								.exchangeStart(Date.from(aei.getExchangeStart()))
+								.exchangeStart(Date.from(aei.getExchangeStart()))
+								.initiator(toInitiatorEnumAsk(aei.getInitiator()))
+								.exchangeEnd(Date.from(aei.getExchangeEnd())).status(aei.getStatus().toString())
+								.failedMessage(aei.getFailedMessage()))
+						.collect(Collectors.toList());
+
+				AskResult ar = new AskResult().bindingSet(this.bindingSetToList(askResult.getBindings()))
```

Comment:
I would expect the knowledge gaps to be included in the `AskResult` here to be returned to the user of the REST API. Or do you return those in some other way?

Reply:
This is my next step...